Frameworks/CoreGraphics/D2DWrapper.mm

```diff
@@ -20,15 +20,21 @@
 using namespace Microsoft::WRL;
 
 // Private helper for creating a D2DFactory
-static ComPtr<ID2D1Factory> __createD2DFactory() {
-    ComPtr<ID2D1Factory> d2dFactory;
-    FAIL_FAST_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, __uuidof(ID2D1Factory), &d2dFactory));
-    return d2dFactory;
+static HRESULT __createD2DFactory(ID2D1Factory** factory) {
+    return D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, factory);
```

Comment:
If we are not using a magic static, there is no real point in having this be a separate function